### PR TITLE
PR for rootfs & raw disk image's reproducible build

### PR DESCRIPTION
**Pull request by Xynnn007**
rootfs, or so-called OS image reproducible build materials. Including

* RBCI's Dockerfile
* rootfs builder
* kata-containers.img (OS image) builder
* OS image checker
* readme

### CODE REVIEW

File: rbi/rootfs-img/rootfs/patch/rootfs-builder/centos/Dockerfile [hunk 1]
```diff
@@ -0,0 +1,44 @@
+FROM registry.centos.org/centos:7
+
+LABEL maintainer="Ding Ma <ding.ma@linux.alibaba.com>"
+
+RUN yum -y update && yum install -y \
+    autoconf-2.69-11.el7.noarch \
```
Comment (hustliyilin) [inline rbi/rootfs-img/rootfs/patch/rootfs-builder/centos/Dockerfile]: use `yum install -y autoconf-2.69` is enough. You can refer to [this dockerfile](https://github.com/alibaba/inclavare-containers/blob/master/tools/docker/Dockerfile-inclavare-dev-alinux2)

图片

File: rbi/rootfs-img/image/patch/Dockerfile [hunk 2]
```diff
@@ -0,0 +1,13 @@
+FROM registry.fedoraproject.org/fedora:34-x86_64
+
+LABEL maintainer="Ding Ma <ding.ma@linux.alibaba.com>"
+
+RUN [ -n "$http_proxy" ] && sed -i '$ a proxy='$http_proxy /etc/dnf/dnf.conf ; true
+
+RUN dnf install -y qemu-img-2:5.2.0-8.fc34.x86_64 \
```
Comment (hustliyilin) [inline rbi/rootfs-img/image/patch/Dockerfile]: ditto

File: rbi/rootfs-img/image-check/Dockerfile [hunk 3]
```diff
@@ -0,0 +1,20 @@
+FROM registry.fedoraproject.org/fedora:34-x86_64
+
+LABEL maintainer="Ding Ma <ding.ma@linux.alibaba.com>"
+
+RUN [ -n "$http_proxy" ] && sed -i '$ a proxy='$http_proxy /etc/dnf/dnf.conf ; true
+
+RUN dnf install -y qemu-img-2:5.2.0-8.fc34.x86_64 \
```
Comment (hustliyilin) [inline rbi/rootfs-img/image-check/Dockerfile]: ditto

File: rbi/rootfs-img/rootfs/.gitignore [hunk 4]
```diff
@@ -1,3 +1,4 @@
 kata-Centos-dnf.conf
 rootfs
-report
\ No newline at end of file
+report
```
Comment (hustliyilin) [inline rbi/rootfs-img/rootfs/.gitignore:3]: You should write `.gitignore` in the lexicographical order. So please change to

```shell
bin
kata-Centos-dnf.conf
report
rootfs
```

File: rbi/rootfs-img/rootfs/patch/rootfs-builder/centos/Dockerfile [hunk 5]
```diff
@@ -2,32 +2,260 @@ FROM registry.centos.org/centos:7
 
 LABEL maintainer="Ding Ma <ding.ma@linux.alibaba.com>"
 
-RUN yum -y update && yum install -y \
-    autoconf-2.69-11.el7.noarch \
-    automake-1.13.4-3.el7.noarch \
+RUN yum update -y \
```
Comment (hustliyilin) [inline rbi/rootfs-img/rootfs/patch/rootfs-builder/centos/Dockerfile]: ditto

File: rbi/rootfs-img/rootfs/patch/rootfs-builder/centos/Dockerfile [hunk 6]
```diff
@@ -2,32 +2,260 @@ FROM registry.centos.org/centos:7
...
-    git-1.8.3.1-23.el7_8.x86_64 \
-    glibc-common-2.17-324.el7_9.x86_64 \
-    glibc-devel-2.17-324.el7_9.x86_64 \
-    glibc-headers-2.17-324.el7_9.x86_64 \
-    glibc-static-2.17-324.el7_9.x86_64 \
-    glibc-utils-2.17-324.el7_9.x86_64 \
-    libseccomp-2.3.1-4.el7.x86_64 \
-    libseccomp-devel-2.3.1-4.el7.x86_64 \
-    libstdc++-devel-4.8.5-44.el7.x86_64 \
+    fontconfig-2.13.0-4.3.el7.x86_64 \
```
Comment (hustliyilin) [inline rbi/rootfs-img/rootfs/patch/rootfs-builder/centos/Dockerfile]: ditto

File: rbi/rootfs-img/image/build-img.sh [hunk 7]
```diff
@@ -88,7 +89,7 @@ main() {
     local abs_source_code_dir=$(cd "$source_code_dir";pwd)
     local abs_output_dir=$(cd "$output_dir";pwd)
     local abs_rootfs_dir=$(cd "$rootfs_dir";pwd)
-    abs_pwd=$(pwd)
+
```
Comment (hustliyilin) [inline rbi/rootfs-img/image/build-img.sh:92]: delete the empty line please

File: rbi/rootfs-img/image-check/Dockerfile (hunk 3 above)
Comment (Xynnn007) [reply rbi/rootfs-img/image-check/Dockerfile]: Those version info has been deleted. And version format in `rootfs-img/rootfs/patch/rootfs-builder/centos/config.sh` has been changed too.

File: rbi/rootfs-img/rootfs/.gitignore (hunk 4 above)
Comment (Xynnn007) [reply rbi/rootfs-img/rootfs/.gitignore]: deleted useless .gitignore s , and rewrite top-level .gitignore in the lexicographical order.

File: rbi/rootfs-img/image/build-img.sh (hunk 7 above)
Comment (Xynnn007) [reply rbi/rootfs-img/image/build-img.sh:92]: I have rechecked all the shells and delete 3 empty lines.

File: rbi/rootfs-img/rootfs/.gitignore (hunk 4 above)
Comment (Xynnn007) [reply rbi/rootfs-img/rootfs/.gitignore]: https://github.com/alibaba/inclavare-containers/pull/1255/commits/2f8c530cfc93ded8b7c66fe8e5a87492beac1089

File: rbi/rootfs-img/image/build-img.sh (hunk 7 above)
Comment (Xynnn007) [reply rbi/rootfs-img/image/build-img.sh:92]: https://github.com/alibaba/inclavare-containers/pull/1255/commits/2ac58355fc80d4fc109ab6fe8b66b3f388dbb973

File: rbi/rootfs-img/image-check/Dockerfile (hunk 3 above)
Comment (Xynnn007) [reply rbi/rootfs-img/image-check/Dockerfile]: https://github.com/alibaba/inclavare-containers/pull/1255/commits/838ef63bc149424ba4f60150d075ab633f523673

File: rbi/rootfs-img/rootfs/patch/rootfs-builder/centos/Dockerfile (hunk 1 above)
Comment (Xynnn007) [reply rbi/rootfs-img/rootfs/patch/rootfs-builder/centos/Dockerfile]: as mentioned below

File: rbi/rootfs-img/rootfs/patch/rootfs-builder/centos/Dockerfile (hunk 5 above)
Comment (Xynnn007) [reply rbi/rootfs-img/rootfs/patch/rootfs-builder/centos/Dockerfile]: mentioned

Comment (Xynnn007) [reply rbi/rootfs-img/rootfs/patch/rootfs-builder/centos/Dockerfile]: mentioned

File: rbi/rootfs-img/rootfs/patch/rootfs-builder/centos/Dockerfile (hunk 6 above)
Comment (Xynnn007) [reply rbi/rootfs-img/rootfs/patch/rootfs-builder/centos/Dockerfile]: mentioned

**Review by hustliyilin — APPROVED** [review]

**Pull request merged**